File: solutions/c/darts/1/src/darts.c
#include "darts.h"

#include <math.h>

uint8_t
score(coordinate_t landing_position) {
  float dist = sqrt(powf(landing_position.x, 2) + powf(landing_position.y, 2));
  if(dist <= 1)
    return 10;
  else if(dist <= 5)
    return 5;
  else if(dist <= 10)
    return 1;
  else
    return 0;
}

File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position :: position) :: integer
  def score({x, y}) do
    r = :math.sqrt(x*x + y*y)
    cond do
      r <= 1 -> 10
      r <= 5 -> 5
      r <= 10 -> 1
      true -> 0
    end
  end
end
